spec/command-registry-spec.js

```diff
@@ -357,12 +358,39 @@ describe("CommandRegistry", () => {
       expect(called).toBe(true);
     });
 
-    it("returns a boolean indicating whether any listeners matched the command", () => {
+    it("returns a promise if any listeners matched the command", () => {
       registry.add('.grandchild', 'command', () => {});
 
-      expect(registry.dispatch(grandchild, 'command')).toBe(true);
-      expect(registry.dispatch(grandchild, 'bogus')).toBe(false);
-      expect(registry.dispatch(parent, 'command')).toBe(false);
+      expect(registry.dispatch(grandchild, 'command').constructor.name).toBe("Promise");
+      expect(registry.dispatch(grandchild, 'bogus')).toBe(null);
+      expect(registry.dispatch(parent, 'command')).toBe(null);
+    });
+
+    it("returns a promise that resolves when the listeners resolve", async () => {
+      registry.add('.grandchild', 'command', () => 1);
+      registry.add('.grandchild', 'command', () => Promise.resolve(2));
+      registry.add('.grandchild', 'command', () => new Promise((resolve) => {
```

Comment:
Why not just call `Promise.resolve` here?

Reply:
I'm trying to make sure the call to dispatch won't resolve before all promises are fulfilled in the future.